### PR TITLE
display correct page when user click on differnt button on the invoic…

### PR DESCRIPTION
**Pull request by Chee-Ng**
…e contact detail page

write to db when user submit

### CODE REVIEW

File: app/models/facilities_management/procurement_contact_detail.rb [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+    validates :organisation_address_line_1, presence: true, format: { with: /\A([a-zA-Z\- (0-9)]*)\z/ }, on: :new_invoicing_address
+    validates :organisation_address_town, presence: true, format: { with: /\A([a-zA-Z\- ]*)\z/ }, on: :new_invoicing_address
+    validates :organisation_address_postcode, presence: true, format: { with: /\A([a-zA-Z (0-9)]*)\z/ }, on: :new_invoicing_address && :new_invoicing_contact_details
+    def contact_address
+      "#{organisation_address_line_1}#{', ' + organisation_address_line_2 if organisation_address_line_2&.present?}#{', ' + organisation_address_town}#{', ' + organisation_address_county if organisation_address_county&.present?}#{', ' + organisation_address_postcode}"
+    end
 
-    def full_organisation_address
-      "#{organisation_address_line_1}#{', ' + organisation_address_line_2 if organisation_address_line_2.present?}#{', ' + organisation_address_town}#{', ' + organisation_address_county if organisation_address_county.present?}#{', ' + organisation_address_postcode}"
+    def valid_postcode?
```
Comment (iceraluk) [inline app/models/facilities_management/procurement_contact_detail.rb]: feels like this should be called valid_contact_address?

File: app/models/facilities_management/procurement_contact_detail.rb [hunk 2]
```diff
@@ -1,15 +1,18 @@
 module FacilitiesManagement
```
Comment (iceraluk) [inline app/models/facilities_management/procurement_contact_detail.rb]: some unit tests for this model would be useful

File: app/models/facilities_management/procurement_contact_detail.rb (hunk 1 above)
Comment (Chee-Ng) [reply app/models/facilities_management/procurement_contact_detail.rb]: Fixed and thanks

File: app/models/facilities_management/procurement_contact_detail.rb (hunk 2 above)
Comment (Chee-Ng) [reply app/models/facilities_management/procurement_contact_detail.rb]: This has been done but not with this commit https://github.com/Crown-Commercial-Service/crown-marketplace/pull/1314

**Comment by ManFromMons** [comment]
This work is also resolved and PR not required

**Pull request closed without merging**